chroma-manager/chroma_core/plugins/linux.py

```diff
@@ -175,23 +181,46 @@ def __init__(self, resource_manager, scannable_id=None):
         super(Linux, self).__init__(resource_manager, scannable_id)
 
         self.major_minor_to_node_resource = {}
+        self.current_devices = "{}"
 
     def teardown(self):
-        self.log.debug("Linux.teardown")
+        log.debug("Linux.teardown")
 
-    def agent_session_continue(self, host_id, data):
+    def agent_session_continue(self, host_id, _):
         # The agent plugin sends us another full report when it thinks something has changed
-        self.agent_session_start(host_id, data, initial_scan=False)
+        self.agent_session_start(host_id, None, initial_scan=False)
 
-    def agent_session_start(self, host_id, data, initial_scan=True):
-        devices = data
+    def agent_session_start(self, host_id, _, initial_scan=True):
+        # devices = data
         initiate_device_poll = False
         reported_device_node_paths = []
 
-        for expected_item in ['vgs', 'lvs', 'emcpower', 'zfspools', 'zfsdatasets', 'zfsvols', 'mpath', 'devs', 'local_fs', 'mds']:
-            if expected_item not in devices:
+        # todo: error handling
+        fqdn = ManagedHost.objects.get(id=host_id).fqdn
+        devices = get_block_devices(fqdn)
+
+        # todo: EMCPower Device detection has been deprecated and mpath it is not provided and is unused
```

Comment:
Mpath still needed, assume it's added back in later commits.

Reply:
yes it is